core/domain/feedback_services_test.py

```diff
@@ -64,4 +65,188 @@ def test_status_of_newly_created_thread_is_open(self):
+            self.assertEqual(thread.status, 
+                             feedback_models.STATUS_CHOICES_OPEN)
+            self.assertEqual(suggestion.exploration_id, self.EXP_ID3)
+            self.assertEqual(suggestion.author_id, 'author_id')
+            self.assertEqual(suggestion.exploration_version, 3)
+            self.assertEqual(suggestion.state_name, 'state_name')
+            self.assertEqual(suggestion.state_content, {'old_content': {}})
+
+    def _get_threads(self, exploration_id):
+        return self.THREADS
+
+    def test_get_open_threads_with_suggestions(self):
+        with self.swap(feedback_models.FeedbackThreadModel, 'get_threads',
+                       self._get_threads):
+            threads = feedback_services.get_open_threads(
+                self.EXP_ID1, True)
+        self.assertEqual(len(threads), 1)
+        self.assertEqual(
+            threads[0].id, '.'.join([self.EXP_ID1, self.THREAD_ID1]))
+
+    def test_get_open_threads_without_suggestions(self):
+        with self.swap(feedback_models.FeedbackThreadModel, 'get_threads',
+                       self._get_threads):
+            threads = feedback_services.get_open_threads(
+                self.EXP_ID1, False)
+        self.assertEqual(len(threads), 1)
+        self.assertEqual(
+            threads[0].id, '.'.join([self.EXP_ID1, self.THREAD_ID5]))
+
+    def test_get_closed_threads_with_suggestions(self):
+        with self.swap(feedback_models.FeedbackThreadModel, 'get_threads',
+                       self._get_threads):
+            threads = feedback_services.get_closed_threads(
+                self.EXP_ID1, True)
+        self.assertEqual(len(threads), 2)
+        self.assertEqual(
+            threads[0].id, '.'.join([self.EXP_ID1, self.THREAD_ID2]))
+        self.assertEqual(
+            threads[1].id, '.'.join([self.EXP_ID1, self.THREAD_ID3]))
+
+    def test_get_closed_threads_without_suggestions(self):
+        with self.swap(feedback_models.FeedbackThreadModel, 'get_threads',
+                       self._get_threads):
+            threads = feedback_services.get_closed_threads(
+                self.EXP_ID1, False)
+        self.assertEqual(len(threads), 1)
+        self.assertEqual(
+            threads[0].id, '.'.join([self.EXP_ID1, self.THREAD_ID4]))
+
+    def test_get_all_suggestion_threads(self):
+        with self.swap(feedback_models.FeedbackThreadModel, 'get_threads',
+                       self._get_threads):
+            threads = feedback_services.get_all_suggestion_threads(
+                self.EXP_ID1)
+        self.assertEqual(len(threads), 3)
+        self.assertEqual(
+            threads[0].id, '.'.join([self.EXP_ID1, self.THREAD_ID1]))
+        self.assertEqual(
+            threads[1].id, '.'.join([self.EXP_ID1, self.THREAD_ID2]))
+        self.assertEqual(
+            threads[2].id, '.'.join([self.EXP_ID1, self.THREAD_ID3]))
+ 
+    def _return_true(self, thread_id, exploration_id):
```

Comment:
Any particular reason for this. I'm asking because this has the added overhead of creating a common parent for these 2 classes with the common setup in it.